src/redux/shoppingCart.js

```diff
@@ -69,8 +70,21 @@ const shoppingCartReducer = createSlice({
 
       saveCartToLocalStorage(state);
     },
+    addToWishlist(state, action) {
+      const { album } = action.payload;
+      console.log('Album ID to add to wishlist:', album.id);
+      axiosInstance
+        .post('/wishlist/', { albumId: album.id })
```

Comment:
The `POST /api/v1/wishlist` endpoint that you're hitting here is a sort of get-or-create endpoint: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/926a305b7f6be72658f74b652b10b02cb33696b2/src/controllers/wishlistController.js#L6](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/926a305b7f6be72658f74b652b10b02cb33696b2/src/controllers/wishlistController.js#L6)

If the user doesn't already have a wishlist, it creates a new empty one. If they already had a wishlist, it returns that wishlist in the response. 

So we do want to make this call, but we can't pass the `albumId` here. Instead, we then must make a second call to add the album to the wishlist using the add album to wishlist endpoint: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/1e48ca722a1a852752cda040cae126beeffd535d/src/routes/wishlistRoutes.js#L23](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/1e48ca722a1a852752cda040cae126beeffd535d/src/routes/wishlistRoutes.js#L23)

A future improvement we could make is to have the user's wishlist be returned as part of the user info as soon as they are logged in or somewhere else earlier/higher up in the component chain, so that here we would already have the wishlist id and could go straight to adding the album. But for now, doing it this way is totally reasonable :smile:

Reply:
@akosasante  I wrote the logic in handleMoveToWishlist in OrderSidebar.js instead of shoppingCart.js because Redux doesn't allow directly using asynchronous functions. However, it seems we can achieve this through the Redux Toolkit library. I'm not very familiar with it, but if it's the right solution, I can give it a try. Or am I reasoning incorrectly?